### DISCUSSION TITLE
Segment to display current ssh host name

### DISCUSSION THREAD

**Discussion by davidshen84** [erikw/tmux-powerline#448] [Q&A] [answered]
The `hostname` segment only shows the hostname of the machine from which the tmux is launched. If I ssh to another host, the hostname won't update.

Do we have a segment to display the ssh host's name?

**Comment by xx4h** [comment, marked as answer]
Hi @davidshen84,

you're right, `hostname` segment is just for displaying the hostname of the system the tmux session is running on.
E.g. if you have multiple tmux sessions, running on different systems, you can quickly differentiate between them by the hostname.

We don't have a segment right now for what you want to achieve, but i guess we could try create one. As a first collection of ideas:
1. Get `tmux display -p -F "#{pane_current_command}"` and check for `ssh`
2. If we've a `ssh` command, we get the `pid` of the pane with `tmux display -p -F "#{pane_pid}"`
3. Based on that `pid` we get all the child PIDs like `ps --no-headers --ppid=PANE_PID -o pid,comm` and parse out the ssh PID
4. We retrieve information about the ssh connection with `ps --no-headers -q SSH_PID -o args` and parse out some information like remote address (e.g. we could reverse lookup), etc.

To get the `hostname` from inside the remote ssh host itself could be a bit of a challenge though.

**Reply by erikw** [reply]
Yes, @davidshen84 feel free to creaet a segment for ssh hostname if you want :)
